crypto/crmf/crmf_lib.c

```diff
@@ -0,0 +1,769 @@
+        if (sig->poposkInput != NULL) {
+/* According to RFC 4211:
+publicKey contains a copy of the public key from the certificate template.
+This MUST be exactly the same value as is contained in the certificate template.
+*/
+            if (pubkey == NULL ||
+                sig->poposkInput->publicKey == NULL ||
+                X509_PUBKEY_cmp(pubkey, sig->poposkInput->publicKey) ||
+                ASN1_item_verify(ASN1_ITEM_rptr(OSSL_CRMF_POPOSIGNINGKEYINPUT),
+                                 sig->algorithmIdentifier, sig->signature,
+                                 sig->poposkInput,
+                                 X509_PUBKEY_get0(pubkey)) < 1)
+                break;
+        } else {
+            if (pubkey == NULL ||
+                req->certReq->certTemplate->subject == NULL ||
+                ASN1_item_verify(ASN1_ITEM_rptr(OSSL_CRMF_CERTREQUEST),
+                                 sig->algorithmIdentifier, sig->signature,
+                                 req->certReq,
+                                 X509_PUBKEY_get0(pubkey)) < 1)
+                break;
+        }
+        return 1;
+    case OSSL_CRMF_POPO_KEYENC:
+        if (req->popo->value.keyEncipherment->type
+            != OSSL_CRMF_POPOPRIVKEY_SUBSEQUENTMESSAGE)
+            goto unsupported;
+        if (ASN1_INTEGER_get
+            (req->popo->value.keyEncipherment->value.subsequentMessage) !=
+            OSSL_CRMF_SUBSEQUENTMESSAGE_ENCRCERT)
+            goto unsupported;
+#if 0 /* TODO enable code when implemented in CMP_certrep_new() */
+        srv_ctx->encryptcert = 1;
+        return 1;
+#else
+        goto unsupported;
+#endif
+    case OSSL_CRMF_POPO_KEYAGREE:
+    default:
+    unsupported:
+        CRMFerr(CRMF_F_OSSL_CRMF_MSGS_VERIFY_POPO,
+                CRMF_R_UNSUPPORTED_POPO_METHOD);
+        return 0;
+    }
+    CRMFerr(CRMF_F_OSSL_CRMF_MSGS_VERIFY_POPO,
+            CRMF_R_UNSUPPORTED_POPO_NOT_ACCEPTED);
+    return 0;
+}
+
+/* retrieves the serialNumber of the given cert template or NULL on error */
+ASN1_INTEGER *OSSL_CRMF_CERTTEMPLATE_get0_serialNumber(OSSL_CRMF_CERTTEMPLATE *tmpl)
+{
+    return tmpl != NULL ? tmpl->serialNumber : NULL;
+}
+
+/* retrieves the issuer name of the given cert template or NULL on error */
+X509_NAME *OSSL_CRMF_CERTTEMPLATE_get0_issuer(OSSL_CRMF_CERTTEMPLATE *tmpl)
+{
+    return tmpl != NULL ? tmpl->issuer : NULL;
+}
+
+/* fill in certificate template.
+   Any value argument that is NULL will leave the respective field unchanged.
+ */
+int OSSL_CRMF_CERTTEMPLATE_fill(OSSL_CRMF_CERTTEMPLATE *tmpl,
+                                const EVP_PKEY *pubkey,
+                                const X509_NAME *subject,
+                                const X509_NAME *issuer,
+                                const ASN1_INTEGER *serial)
+{
+    if (tmpl == NULL) {
+        CRMFerr(CRMF_F_OSSL_CRMF_CERTTEMPLATE_FILL, CRMF_R_NULL_ARGUMENT);
+        return 0;
+    }
+    if ((pubkey != NULL &&
```

Comment:
&& goes to the next line, usually with 4 spaces extra indentation per our coding style guide

Reply:
in this case, probably better to split the expression in several if(...) goto oom?

Reply:
The coding style needs updating in various ways. This rule should be added (because we do apply that style).

Reply:
This was also done in November:
```
commit 2f301f13961ad855ec36596092c798b763990684
Author: David von Oheimb <David.von.Oheimb@siemens.com>
Date:   Mon Nov 26 09:58:42 2018 +0100

    fix various minor conding style issues (regarding layout etc.)
```